Leetcode/aoc_2015_d6.py

```diff
@@ -100,6 +121,34 @@ def part_one(instructions: list) -> int:
 At first, I considered the fact that we can just use math, on add 1, off sub 1, tog add 2.
 But then I realized that wont be possible since the minimum brightness is zero, can't go under.
 '''
+
+def inc_brightness(grid: list[list], r1: int, r2: int, c1: int, c2: int):
+	for row in range(r1, r2 + 1):
+		for col in range(c1, c2 + 1):
+			grid[row][col] += 1
+
+
+def dec_brightness(grid: list[list], r1: int, r2: int, c1: int, c2: int):
+	for row in range(r1, r2 + 1):
+		for col in range(c1, c2 + 1):
+			if grid[row][col] > 0:
+				grid[row][col] -= 1
+
+
+def tog_brightness(grid: list[list], r1: int, r2: int, c1: int, c2: int):
+	for row in range(r1, r2 + 1):
+		for col in range(c1, c2 + 1):
+			grid[row][col] += 2
```

Comment:
Similarly here, note these all just apply deltas (increments/decrements) to the existing brightness: so just one function `add_brightness(..., amount: int)` seems fine here. Then you  can do +1, -1, and +2 respectively.

Reply:
Makes sense, just like the part one. Changed.